### PR TITLE
Support passing array of commands into given method

### PR DESCRIPTION
**Pull request by rhyslbw**
This change makes it possible to test more complex domain models by allowing multiple commands to setup state.  I've added a unit test for `AggregateTest` but since the symbol is not exported I'm not sure how to gain access without attaching it to the global scope, so have left it commented out.

screen shot 2016-02-14 at 19 50 56

### CODE REVIEW

File: source/aggregates-bdd-api.coffee [hunk 1]
```diff
@@ -22,16 +22,35 @@ class AggregateTest
...
-      # We just send the command through the app and let
+    if data is undefined then return this
+    commands = []
+    events = []
+    if !_.isArray(data)
+      if data instanceof Space.messaging.Event
+        events.push(data);
+      if data instanceof Space.messaging.Command
+        commands.push(data)
+    else if _.isArray(data)
```
Comment (DominikGuzei) [inline source/aggregates-bdd-api.coffee]: This check is not necessary as it must be an array in that case ;-)

File: source/aggregates-bdd-api.coffee [hunk 2]
```diff
@@ -22,16 +22,35 @@ class AggregateTest
...
-      changes = events: data, commands: []
-      aggregateId = data[0].sourceId
-      version = data[0].version ? 1
-      @_commitStore.add changes, aggregateId, version - 1
-    else if data instanceof Space.messaging.Command
-      # We just send the command through the app and let
+    if data is undefined then return this
+    commands = []
+    events = []
+    if !_.isArray(data)
```
Comment (DominikGuzei) [inline source/aggregates-bdd-api.coffee:28]: Maybe you could just wrap the data into an array like `data = [data]` when it's not an array :wink: so you just need the array handling code to setup the commands / events.

Comment (rhyslbw) [inline source/aggregates-bdd-api.coffee:28]: @DominikGuzei Good call, changes in latest commit

**Comment by rhyslbw** [comment]
@DominikGuzei please check this out when possible. It's only WIP as we need to decide if it's ok to proceed without the test being included in the suite. Post Meteor 1.3 we should have no problem

**Comment by DominikGuzei** [comment]
Which symbol is not exported? i don't fully understand the problem :wink:
I would just write the test in a way that works with < Meteor 1.3 for now … since that is our target.

**Comment by rhyslbw** [comment]
@DominikGuzei 

> I would just write the test in a way that works with < Meteor 1.3 for now … since that is our target.

Guess we should just export `AggregateTest` from the package then?

**Pull request closed without merging**